Introduced Amcrest camera sensors and bump Amcrest module version

tchellomello opened the pull request:
**Description:**
This PR introduces 3 sensors for the Amcrest cameras.  I'm still working to add more sensors soon. 

This change also bumps the version of Amcrest python module. 

* Badges
![amcrest_sensor_badges](https://cloud.githubusercontent.com/assets/809840/21950686/8410d6de-d9cb-11e6-9c3b-12305a75ccd2.png)

* Motion Card
![motion](https://cloud.githubusercontent.com/assets/809840/21950734/dfc076a6-d9cb-11e6-85d3-31adb58ce5c1.png)

* Preset card
![preset](https://cloud.githubusercontent.com/assets/809840/21950735/dfc20692-d9cb-11e6-8474-df3e190acf09.png)

* SD card
![sd_card](https://cloud.githubusercontent.com/assets/809840/21950736/dfc1efea-d9cb-11e6-953c-dad87dac0890.png)

**Pull request in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io) with documentation (if applicable):** home-assistant/home-assistant.github.io#1763

**Example entry for `configuration.yaml` (if applicable):**
```yaml
# sensors.yaml
- platform: amcrest
  name: "Amcrest Living Room"
  host: !secret amcrest_living_room_host
  username: !secret amcrest_living_room_username
  password: !secret amcrest_living_room_password
  monitored_conditions:
    - motion_detector
    - sdcard
    - ptz_preset
```

**Checklist:**

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io)

If the code communicates with devices, web services, or third-party tools:
  - [x] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [x] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
  - [x] New dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
  - [x] New files were added to `.coveragerc`.

tchellomello committed:
Introduced Amcrest camera sensors

tchellomello committed:
Makes script/gen_requirements_all.py happy

tchellomello committed:
Bump Amcrest version across all components

pvizeli commented on homeassistant/components/sensor/amcrest.py:
```diff
@@ -0,0 +1,144 @@
+    for sensor_type in config.get(CONF_MONITORED_CONDITIONS):
+        sensors.append(AmcrestSensor(config, data, sensor_type))
+
+    add_devices(sensors)
```
use `add_devices(sensors, True)` and remove update from init.

pvizeli commented on homeassistant/components/sensor/amcrest.py:
```diff
@@ -0,0 +1,144 @@
+    try:
+        data.camera.current_time
+    # pylint: disable=broad-except
+    except Exception as ex:
```
Only catch defined exceptions on this point.

pvizeli commented on homeassistant/components/sensor/amcrest.py:
```diff
@@ -0,0 +1,144 @@
+        """Return the units of measurement."""
+        return SENSOR_TYPES.get(self._sensor_type)[1]
+
+    @Throttle(MIN_TIME_BETWEEN_UPDATES)
```
Is not needed and have no effect there. You can use `SCAN_INTERVAL` on top of you platform and user can overwrite that with `scan_interval` in platform config.

If you don't want to hit your camera, implement this on your data object.

tchellomello commented:
@pvizeli  Thanks for the review. I'll update the PR soon.

tchellomello committed:
- Adjusted scan_interval to 10 seconds

- Filtering HTTPError and ConnectTimeout exceptions

- Removed @Throttle decorator

tchellomello commented:
@pvizeli  PR updated. Please let me know if looks good now. Thank you!
